File: code/chapter_guess_the_number/guess_the_number.rb
puts "Guess The Number!"
puts "----"

puts "I can guess a number between 1 and any number you give me."
puts "What's the highest number I should pick?"

number = gets

puts "Ok, I will guess a number between 1 and " + number
number = number.to_i - 1

computer_number =  rand(number.to_i) + 1

winner = false

while winner == false
  puts "Ok, guess the number!"
  guess = gets.to_i
  if guess == computer_number
    puts "You got it!"
    winner = true
  else
    if guess < computer_number
      puts "Too low. Try again!"
    else
      puts "Too high. Try again!"
    end
  end
end

File: code/chapter_guess_the_number/guess_the_number.py
# -*- coding: utf-8 -*-

import random

print "Guess The Number!"
print "----"

print "I can guess a number between 1 and any number you give me."

while True:
	try:
		number = raw_input( "What's the highest number I should pick? " )
		number = int( number )
		break
	except ValueError, e:
		print "\"%s\" isn't a number!" % number

print "Ok, I will guess a number between 1 and", number

computer_number =  random.randint( 0, number )

winner = False

while winner == False:
	while True:
		try:
			guess = raw_input( "Ok, guess the number! " )
			guess = int( guess )
			break
		except ValueError, e:
			print "\"%s\" isn't a number!" % guess
	if guess == computer_number:
		print "You got it!"
		winner = True
	else:
		if guess < computer_number:
			print "Too low. Try again!"
		else:
			print "Too high. Try again!"

